student_class.rb

```diff
@@ -1,15 +1,13 @@
-require './personClass'
+require './person_class'
 
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+  def initialize(age, classroom, name = 'Unknown')
+    super(age, name)
     @classroom = classroom
+    classroom.add_student(self)
```

Comment:
- [ ] The Student class should have its own method for associating with the classroom also.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```